### PR TITLE
Add acceptance test for watch query changed via mutation

### PR DESCRIPTION
**Pull request by chadian**
I wrote a test to better understand how `watchQuery` should work if a mutation is performed. Maybe it's valuable to be added to the test suite? I could refactor it into a route unit test, too. Cheers ✌🏽

### CODE REVIEW

File: tests/acceptance/watch-query-test.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        human(obj, args) {
+          assert.deepEqual(args, { id: '1000' });
+          return Object.assign({}, character);
+        },
+      },
+      Mutation: {
+        changeCharacterName(_, { id, name }) {
+          assert.equal(id, mockHuman.id, 'correct mutuation id is passed in');
+          assert.equal(name, 'Han Solo', 'correct mutuation id is passed in');
+          return Object.assign(character, { name });
```
Comment (bgentry) [inline tests/acceptance/watch-query-test.js]: On line 28 above, you're making a copy of `character` into a new `Object`, but here you're mutating `character`. I'd be less worried about side effects if this was instead:

```js
return Object.assign({}, character, { name });
```

File: tests/acceptance/watch-query-test.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      Query: {
+        human(obj, args) {
+          assert.deepEqual(args, { id: '1000' });
+          return Object.assign({}, character);
+        },
+      },
+      Mutation: {
+        changeCharacterName(_, { id, name }) {
+          assert.equal(id, mockHuman.id, 'correct mutuation id is passed in');
+          assert.equal(name, 'Han Solo', 'correct mutuation id is passed in');
```
Comment (bgentry) [inline tests/acceptance/watch-query-test.js]: s/mutation id/name

Comment (bgentry) [reply tests/acceptance/watch-query-test.js]: this example should totally start as "Anakin Skywalker" and then mutate its name to "Darth Vader" :joy:

**Comment by chadian** [comment]
Great suggestions, also Anakin → Darth 😁
I've made the requested changes, let me know if they line up with what you're after.
Cheers!

**Comment by bgentry** [comment]
Thanks @chadian :v:

**Pull request merged**